nac/workflows/workflow_coop.py

```diff
@@ -0,0 +1,119 @@
+"""Crystal Orbital Overlap Population  calculation."""
+__all__ = ['workflow_crystal_orbital_overlap_population']
+
+import logging
+import numpy as np
+from nac.common import (
+    number_spherical_functions_per_atom,
+    retrieve_hdf5_data, is_data_in_hdf5)
+from nac.integrals.multipole_matrices import compute_matrix_multipole
+from nac.workflows.initialization import initialize
+from nac.workflows.workflow_single_points import workflow_single_points
+from scipy.constants import physical_constants
+from qmflows.parsers.xyzParser import readXYZ
+
+# Starting logger
+LOGGER = logging.getLogger(__name__)
+
+def workflow_crystal_orbital_overlap_population(config: dict) -> list:
```

Comment:
This function is very complicated, it would be great if you split it in several smaller pieces that do just a single task